extensions/wrapper/wrapper/src/main/java/de/sovity/edc/ext/wrapper/api/ui/pages/dashboard/DashboardPageApiService.java

```diff
@@ -14,22 +14,52 @@
 
 package de.sovity.edc.ext.wrapper.api.ui.pages.dashboard;
 
+import de.sovity.edc.ext.wrapper.api.ui.model.ContractAgreementDirection;
 import de.sovity.edc.ext.wrapper.api.ui.model.DashboardPage;
+import de.sovity.edc.ext.wrapper.api.ui.model.DashboardTransferAmounts;
 import de.sovity.edc.ext.wrapper.api.ui.pages.dashboard.services.DapsConfigService;
+import de.sovity.edc.ext.wrapper.api.ui.pages.dashboard.services.DashboardDataFetcher;
 import de.sovity.edc.ext.wrapper.api.ui.pages.dashboard.services.MiwConfigService;
 import de.sovity.edc.ext.wrapper.api.ui.pages.dashboard.services.SelfDescriptionService;
+import de.sovity.edc.ext.wrapper.api.ui.pages.transferhistory.TransferProcessStateService;
 import lombok.RequiredArgsConstructor;
+import org.eclipse.edc.connector.transfer.spi.types.TransferProcess;
 import org.jetbrains.annotations.NotNull;
 
+import java.util.List;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import static de.sovity.edc.ext.wrapper.api.ui.model.TransferProcessSimplifiedState.ERROR;
+import static de.sovity.edc.ext.wrapper.api.ui.model.TransferProcessSimplifiedState.OK;
+import static de.sovity.edc.ext.wrapper.api.ui.model.TransferProcessSimplifiedState.RUNNING;
+
 @RequiredArgsConstructor
 public class DashboardPageApiService {
+    private final DashboardDataFetcher dashboardDataFetcher;
+    private final TransferProcessStateService transferProcessStateService;
     private final DapsConfigService dapsConfigService;
     private final MiwConfigService miwConfigService;
     private final SelfDescriptionService selfDescriptionService;
 
     @NotNull
     public DashboardPage dashboardPage() {
-        var dashboardPage = new DashboardPage();
+
+        var transferProcesses = dashboardDataFetcher.getTransferProcesses();
+        var negotiations = dashboardDataFetcher.getAllContractNegotiations();
+
+        var providingAgreements = negotiations.stream()
+                .filter(negotiation -> ContractAgreementDirection.fromType(negotiation.getType()).equals(ContractAgreementDirection.PROVIDING))
+                .map(negotiation -> negotiation.getContractAgreement().getId())
+                .collect(Collectors.toSet());
```

Comment:
requires a .filter(Objects::nonNull), because not all negotiations will have contract agreements (add to the test, too)